src/item.h

```diff
@@ -107,6 +108,8 @@ class item_category
 
 class item : public JsonSerializer, public JsonDeserializer, public visitable<item>
 {
+    friend item_location;
```

Comment:
The two classes are coupled together. The other option is making the item uid public (which I can do) but this would encourage it's use in code where it's inappropriate.